### ISSUE TITLE
[Bug]: Creating server instance of a custom image referencing the image location to an existing internal image fails

### ISSUE THREAD

**Issue by jstourac** [opendatahub-io/odh-dashboard#2185]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Deploy type

Downstream version (eg. `OpenShift AI 2.4`)

### Version

2.4.RC2

### Current Behavior

If I create a custom image providing the image location with the link to the internal registry (Generic data science 2023.2 image in this case), the custom image is added successfully but when I then try to invoke a server instance from that image, I get stuck on the `Error: ImagePullBackOff`

How the custom image is created:

![Screenshot from 2023-11-20 13-21-48](https://github.com/opendatahub-io/odh-dashboard/assets/12250881/f208c130-c0ea-4d8f-b2be-d26ef49b2ee7)

Here is the ImageStream of the created custom image:

```
kind: ImageStream
apiVersion: image.openshift.io/v1
metadata:
  annotations:
    opendatahub.io/notebook-image-creator: htpasswd-cluster-admin-user
    opendatahub.io/notebook-image-desc: hello
    opendatahub.io/notebook-image-name: foo
    opendatahub.io/notebook-image-url: >-
      image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/s2i-generic-data-science-notebook:2023.2
    openshift.io/image.dockerRepositoryCheck: '2023-11-20T12:20:52Z'
  resourceVersion: '26576673'
  name: custom-foo
  uid: 6d8e4b5f-2eef-4eaf-bc30-cd46befc95c4
  creationTimestamp: '2023-11-20T12:20:50Z'
  generation: 2
  managedFields:
    - manager: unknown
      operation: Update
      apiVersion: image.openshift.io/v1
      time: '2023-11-20T12:20:50Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            .: {}
            'f:opendatahub.io/notebook-image-creator': {}
            'f:opendatahub.io/notebook-image-desc': {}
            'f:opendatahub.io/notebook-image-name': {}
            'f:opendatahub.io/notebook-image-url': {}
          'f:labels':
            .: {}
            'f:app.kubernetes.io/created-by': {}
            'f:opendatahub.io/dashboard': {}
            'f:opendatahub.io/notebook-image': {}
        'f:spec':
          'f:lookupPolicy':
            'f:local': {}
          'f:tags':
            .: {}
            'k:{"name":"2023.2"}':
              .: {}
              'f:annotations':
                .: {}
                'f:opendatahub.io/notebook-python-dependencies': {}
                'f:opendatahub.io/notebook-software': {}
                'f:openshift.io/imported-from': {}
              'f:from': {}
              'f:generation': {}
              'f:importPolicy':
                .: {}
                'f:importMode': {}
              'f:name': {}
              'f:referencePolicy':
                .: {}
                'f:type': {}
  namespace: redhat-ods-applications
  labels:
    app.kubernetes.io/created-by: byon
    opendatahub.io/dashboard: 'true'
    opendatahub.io/notebook-image: 'true'
spec:
  lookupPolicy:
    local: true
  tags:
    - name: '2023.2'
      annotations:
        opendatahub.io/notebook-python-dependencies: '[]'
        opendatahub.io/notebook-software: '[]'
        openshift.io/imported-from: >-
          image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/s2i-generic-data-science-notebook:2023.2
      from:
        kind: DockerImage
        name: >-
          image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/s2i-generic-data-science-notebook:2023.2
      generation: 2
      importPolicy:
        importMode: Legacy
      referencePolicy:
        type: Source
status:
  dockerImageRepository: >-
    image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/custom-foo
  tags:
    - tag: '2023.2'
      items:
        - created: '2023-11-20T12:20:52Z'
          dockerImageReference: >-
            image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/s2i-generic-data-science-notebook@sha256:2513b3634fc9e19a9905fe42fee030c0f9c354747502120a1a8a66db827a3901
          image: >-
            sha256:2513b3634fc9e19a9905fe42fee030c0f9c354747502120a1a8a66db827a3901
          generation: 2
```

Creation of server instance:

![Screenshot from 2023-11-20 13-27-00](https://github.com/opendatahub-io/odh-dashboard/assets/12250881/8055ec4e-c061-4d52-adaf-375264255732)

### Expected Behavior

I would expect that since the custom image can be created this way, it's possible to invoke a server instance using it.

### Steps To Reproduce

1. Open the RHOAI web UI
2. Go to `Settings -> Notebook image settings`
3. Click on `Import new image` button
4. Fill in the data, eg.:
   * Image location: `image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/s2i-generic-data-science-notebook:2023.2`
   * Name: `foo` 
6. Confirm and go to `Applications -> Enabled` and launch the Jupyter app there
7. Select the `foo` image and click on `Start server` button
8. See the `ImagePullBackOff` error when starting the server

**Comment by shalberd** [comment]
@andrewballantyne FYI

**Comment by andrewballantyne** [comment]
`s2i-generic-data-science-notebook:2023.2` is a OOTB image. I do not believe you can use existing internal registry images like this. cc @harshad16 

I think you want an external registry that gets ingested into the internal registry through our flows. I believe it will always fail if you use the internal registry and try to "duplicate" it? (which is my assumption to what you're currently doing).

Maybe Harshad can help out here.

**Comment by harshad16** [comment]
I think, the test tried above is to check if any image pushed internally can be imported as a custom image from dashboard.
and s2i-generic-data-science is used as an example.

**Comment by andrewballantyne** [comment]
Right, what are you expectations here @harshad16 on if this should ever succeed? I think this is a failure state in OpenShift not in our code, but I could be wrong. I would need to investigate. 

Trying to figure out if this is "by design" or "something we should fix". Do you have an opinion / knowledge on this front here?

**Comment by jstourac** [comment]
Hi guys, thank you all for your inputs.

Harshad is correct - this is one of our test-case scenario. May be related to #1085.

But I need to verify whether it ever worked indeed. It should work with 1.33 but need to check again as there seem to be some unclear expectations on this. Will post here.

**Comment by jstourac** [comment]
Hi guys, an update - I tried manually with the RHODS releases 1.28 - 1.33 (including the never released 1.34) and the truth here is that I wasn't able to make this scenario work in any of the release. Although, per the author of the test this worked quite recently.

Truth is that this scenario may not have much sense in the real world. But at least based on the discussion in https://issues.redhat.com/browse/RHODS-9315 it looks like it was considered there.

@DaoDaoNoCode I can see that you just updated this to your planning. Is this something you're gonna look at then?

**Comment by adnankhan666** [comment]
https://issues.redhat.com/browse/RHOAIENG-1193
